### PR TITLE
refactor primary secondary and sub maneuver

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Refactors the implementation for `Primary`, `Secondary` and `Sub` maneuver and adds the logic into separate components.

### Changelog

```
<changelog></changelog>
```

### Screenshots or Gifs

Note: This PR will be merged only after [this](https://github.com/mapbox/mapbox-navigation-android/pull/4378) will be merged and will automatically merge to `main`

### CODE REVIEW

**Comment by LukasPaczos** [comment]
This is a completely internal refactor, isn't it? Let's skip the changelog entry in this case.

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/internal/ExitStyleGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.style.ImageSpan
+import android.widget.TextView
+import com.mapbox.navigation.ui.utils.internal.extensions.drawableWithHeight
+import com.mapbox.navigation.ui.utils.internal.extensions.getAsBitmap
+
+object ExitStyleGenerator {
+
+    @JvmStatic
```
Comment (LukasPaczos) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/internal/ExitStyleGenerator.kt:16]: We don't need to generate the Jvm static method, this is only an internal class. Same everywhere above.

File: libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/internal/ExitStyleGenerator.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.style.ImageSpan
+import android.widget.TextView
+import com.mapbox.navigation.ui.utils.internal.extensions.drawableWithHeight
+import com.mapbox.navigation.ui.utils.internal.extensions.getAsBitmap
+
+object ExitStyleGenerator {
```
Comment (LukasPaczos) [inline libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/internal/ExitStyleGenerator.kt:14]:
```suggestion
intenral object ExitStyleGenerator {
```

and move out of the `internal` package. Same everywhere below.

**Comment by Guardiola31337** [comment]
Noting that `unit-tests` are ❌

**Review by Guardiola31337 — APPROVED** [review]

**Comment by Guardiola31337** [comment]
Going ahead and merging here 🚀

**Pull request merged**